### ISSUE TITLE
topology.addResidue and Residue.__init__ accept one-letter amino acid codes without warning or error

### ISSUE THREAD

**Issue by jonathanking** [openmm/openmm#3128]
I would like to suggest an improvement to the documentation or input validation for `topology.addResidue` and `Residue.__init__`. The documentation does not point out that for amino acids, it expects 3-letter amino acid codes and not one-letter codes. It's certainly possible that this distinction is made clear elsewhere in the documentation, but it would most likely help to mention this in the documentation for these functions.

I recently solved an issue in my own software where I was adding Residues to a topology object by adding amino acid Residues with the `name` attribute set to their one-letter amino acid codes. This did not cause any issues until I tried to add missing atoms to these residues. My `G` residues were not treated as Glycine as I expected.

**Comment by peastman** [comment]
Topology expects you to use standard PDB names.  G is a [fine residue name](https://ftp.wwpdb.org/pub/pdb/data/monomers/G).  It just happens to be a guanosine base, not glycine.

**Comment by jonathanking** [comment]
Thank you. I did learn that fact about `G` when looking into my code, but it was my mistake to miss the comment in the Topology documentation about using PDB 3.0 nomenclature for residue names. My issue is certainly resolved, but perhaps the documentation for [Residue](http://docs.openmm.org/development/api-python/generated/openmm.app.topology.Residue.html#openmm.app.topology.Residue) could benefit from making this note as well, as that is where I went to learn about what the `name` argument expected.
